### ISSUE TITLE
Unicode substitution broken with helm

### ISSUE THREAD

**Issue by non-Jedi** [JuliaEditorSupport/julia-emacs#204]
@giordano [reports](https://github.com/JuliaEditorSupport/julia-emacs/issues/193#issuecomment-1909226438):

>Yes, "\alph" for me is completed to "\alpha", not "α", further pressing `M-C-i` does nothing. Not sure it's relevant, but I use Helm. I also have auto-complete installed, but doesn't seem the mode is active in my julia buffer.

Locally, I've confirmed that with only `julia-mode` installed, `C-M-i` correctly completes "\alph" to "α". If helm is installed with the following minimal config, `C-M-i` completes "\alph" to "\alpha":

```elisp
(require 'package)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
(package-initialize)

(package-install 'julia-mode)

(package-install 'helm)
(helm-mode 1)
```

This seems like a re-occurrence of https://github.com/emacs-helm/helm/issues/2356, but more investigation is needed.

**Comment by non-Jedi** [comment]
Ran the snippet from https://github.com/emacs-helm/helm/issues/2356, and this issue is not the same as https://github.com/emacs-helm/helm/issues/2356; `:exit-function` is properly called with `'finished`. Will need to do some more digging to figure out what's going on here.

**Comment by non-Jedi** [comment]
Reopening per https://github.com/JuliaEditorSupport/julia-emacs/pull/206#issuecomment-2003889106

**Comment by non-Jedi** [comment]
So the current issue is that `\alph<TAB>` calls `:exit-function` with `'finished` whereas `\alpha<TAB>` calls `:exit-function` with `'exact`. According to the docstring for `completion-extra-properties`, this is a helm bug since the text may not "be further completed". This bug, unlike the one fixed in #206 is a similar but more specific version of https://github.com/emacs-helm/helm/issues/2356.

**Comment by non-Jedi** [comment]
Thierry fixed things on the helm side (https://github.com/emacs-helm/helm/issues/2646). Could you please check again that the issue is fixed for you with latest master of helm @giordano?
